File: lib/mascotConstants.js
/**
 * Flip, Flo & Fins — the swim-coach monkey mascots.
 * Full-body artwork lives in /public/mascot/ with an open-eyes and a
 * closed-eyes (blink) frame per character.
 */
export const MASCOTS = {
  flip: {
    id: 'flip',
    nameKey: 'settings.mascotFlipName',
    descKey: 'settings.mascotFlipDesc',
    rulesKey: 'settings.mascotFlipRules',
    coachedLevel: 'beginner',
    traitKeys: ['mascot.traits.encouraging', 'mascot.traits.approachable'],
    cheerKey: 'mascot.cheerFlip',
    previewKey: 'settings.mascotPreviewFlip',
    // English personality descriptor injected into the AI coach prompt
    aiPersonality: 'encouraging and approachable — celebrates effort, keeps things light, always positive, never criticises',
    images: {
      open: '/mascot/flip-open.png',
      closed: '/mascot/flip-closed.png',
    },
    stageBg: '/mascot/flip-stage-bg.png',
    // natural width / height of the artwork
    aspect: 490 / 900,
    gameplay: {
      challengeIntensity: 0.75, // easier monthly challenge targets
      coinMultiplier: 0.5,      // half coins…
      minSessionCoins: 3,       // …but you can never lose any
      sessionPenalty: false,
      requiredMonthlyTier: null,
      monthlyPenaltyCoins: 0,
      freeMonthlyRerolls: 1,
      positiveOnly: true,
      doubleImprovementBonus: false,
    },
  },
  flo: {
    id: 'flo',
    nameKey: 'settings.mascotFloName',
    descKey: 'settings.mascotFloDesc',
    rulesKey: 'settings.mascotFloRules',
    coachedLevel: 'intermediate',
    traitKeys: ['mascot.traits.friendly', 'mascot.traits.motivating'],
    cheerKey: 'mascot.cheerFlo',
    previewKey: 'settings.mascotPreviewFlo',
    aiPersonality: 'friendly and motivating — warm and enthusiastic, cheers milestones, but honestly points out when statistics decline',
    images: {
      open: '/mascot/flo-open.png',
      closed: '/mascot/flo-closed.png',
      disappointedOpen: '/mascot/flo-disappointed-open.png',
      disappointedClosed: '/mascot/flo-closed.png',
    },
    stageBg: '/mascot/flo-stage-bg.png',
    aspect: 593 / 900,
    gameplay: {
      challengeIntensity: 1,
      coinMultiplier: 1,
      minSessionCoins: 5,
      sessionPenalty: false,
      requiredMonthlyTier: 'silver', // ending a month below silver costs coins
      monthlyPenaltyCoins: 40,
      freeMonthlyRerolls: 1,
      positiveOnly: false,
      doubleImprovementBonus: false,
    },
  },
  fins: {
    id: 'fins',
    nameKey: 'settings.mascotFinsName',
    descKey: 'settings.mascotFinsDesc',
    rulesKey: 'settings.mascotFinsRules',
    coachedLevel: 'advanced',
    traitKeys: ['mascot.traits.focused', 'mascot.traits.challenging'],
    cheerKey: 'mascot.cheerFins',
    previewKey: 'settings.mascotPreviewFins',
    aiPersonality: 'focused and challenging — a demanding performance coach who is direct, critical but fair, always pushes for measurable improvement and calls out slacking',
    images: {
      open: '/mascot/fins-open.png',
      closed: '/mascot/fins-closed.png',
      disappointedOpen: '/mascot/fins-disappointed-open.png',
      disappointedClosed: '/mascot/fins-disappointed-closed.png',
    },
    stageBg: '/mascot/fins-stage-bg.png',
    aspect: 512 / 900,
    gameplay: {
      challengeIntensity: 1.25,  // hardest monthly challenge targets
      coinMultiplier: 1,
      minSessionCoins: null,     // sessions can go negative when you slack
      sessionPenalty: true,
      requiredMonthlyTier: 'gold', // ending a month below gold costs coins
      monthlyPenaltyCoins: 75,
      freeMonthlyRerolls: 2,     // compromise: two rerolls instead of one
      positiveOnly: false,
      doubleImprovementBonus: true,
    },
  },
};

export const MASCOT_IDS = ['flip', 'flo', 'fins'];

export function getMascot(id) {
  return MASCOTS[id] || MASCOTS.flip;
}

import { resolveMascotId } from './mascotUnlock.js';

export { resolveMascotId };

export function getMascotName(id, t) {
  return t(getMascot(id).nameKey);
}

export function getMascotGameplay(id) {
  return getMascot(id).gameplay;
}

/** Coach tier shown on the mascot badge (Flip → beginner, Flo → intermediate, Fins → advanced). */
export function getMascotCoachedLevel(id) {
  return getMascot(id).coachedLevel || 'beginner';
}

export const MASCOT_LEVEL_MAP = {
  advanced: 'advanced',
  intermediate: 'intermediate',
  beginner: 'beginner',
  developing: 'beginner',
  unknown: 'beginner',
};

export function resolveMascotLevel(benchmarkLevel) {
  return MASCOT_LEVEL_MAP[benchmarkLevel] || 'beginner';
}

File: ios/AapSC/Lib/MascotConstants.swift
import Foundation

struct MascotGameplay: Equatable {
    var challengeIntensity: Double
    var coinMultiplier: Double
    var minSessionCoins: Int?
    var sessionPenalty: Bool
    var requiredMonthlyTier: String?
    var monthlyPenaltyCoins: Int
    var freeMonthlyRerolls: Int
    var positiveOnly: Bool
    var doubleImprovementBonus: Bool
}

struct MascotDefinition: Equatable {
    var id: String
    var coachedLevel: String
    var aiPersonality: String
    var gameplay: MascotGameplay
}

enum MascotConstants {
    static let ids = ["flip", "flo", "fins"]

    static let flip = MascotDefinition(
        id: "flip",
        coachedLevel: "beginner",
        aiPersonality: "encouraging and approachable — celebrates effort, keeps things light, always positive, never criticises",
        gameplay: MascotGameplay(
            challengeIntensity: 0.75,
            coinMultiplier: 0.5,
            minSessionCoins: 3,
            sessionPenalty: false,
            requiredMonthlyTier: nil,
            monthlyPenaltyCoins: 0,
            freeMonthlyRerolls: 1,
            positiveOnly: true,
            doubleImprovementBonus: false
        )
    )

    static let flo = MascotDefinition(
        id: "flo",
        coachedLevel: "intermediate",
        aiPersonality: "friendly and motivating — warm and enthusiastic, cheers milestones, but honestly points out when statistics decline",
        gameplay: MascotGameplay(
            challengeIntensity: 1,
            coinMultiplier: 1,
            minSessionCoins: 5,
            sessionPenalty: false,
            requiredMonthlyTier: "silver",
            monthlyPenaltyCoins: 40,
            freeMonthlyRerolls: 1,
            positiveOnly: false,
            doubleImprovementBonus: false
        )
    )

    static let fins = MascotDefinition(
        id: "fins",
        coachedLevel: "advanced",
        aiPersonality: "focused and challenging — a demanding performance coach who is direct, critical but fair, always pushes for measurable improvement and calls out slacking",
        gameplay: MascotGameplay(
            challengeIntensity: 1.25,
            coinMultiplier: 1,
            minSessionCoins: nil,
            sessionPenalty: true,
            requiredMonthlyTier: "gold",
            monthlyPenaltyCoins: 75,
            freeMonthlyRerolls: 2,
            positiveOnly: false,
            doubleImprovementBonus: true
        )
    )

    static func mascot(_ id: String?) -> MascotDefinition {
        switch id {
        case "flo": return flo
        case "fins": return fins
        default: return flip
        }
    }

    static func gameplay(_ id: String?) -> MascotGameplay {
        mascot(id).gameplay
    }

    static func coachedLevel(_ id: String?) -> String {
        mascot(id).coachedLevel
    }
}
